docs/architecture.md

```diff
@@ -0,0 +1,40 @@
+Fluo Architecture
+=================
+
+![fluo-architecture][1]
+
+## Fluo Application
+
+A **Fluo application** maintains a large scale computation using a series of small transactional
+updates.  Multiple Fluo applications can run on a cluster at the same time.  Each Fluo application 
+runs as a Hadoop YARN application and can be stopped, started, and upgraded independently.  Fluo 
+applications consist of an oracle process and a configurable number of worker processes:
+
+ * The **Oracle** process allocates timestamps for transactions.  While only one Oracle is required, 
+   Fluo can be configured to run extra Oracles that can take over if the primary Oracle fails.
+ * **Worker** processes run user code (called **observers**) that perform transactions.  All workers
+   run the same observers.  The number of worker instances are configured to handle the processing 
+   workload.
+ 
+Fluo applications store their data in a **Fluo table** which has a similar structure (row, column, 
+value) to an **Accumulo table** except that a Fluo table has no timestamps.  A Fluo table
+is implemented using an Accumulo table.  While you could scan the Accumulo table used to implement 
+a Fluo table using an Accumulo client, you would read extra implementation-related data in addition
+to your data.  Therefore, developers should only interact with the data in a Fluo table by writing 
+Fluo client or observer code:
+
+ * **Clients** ingest data or interact with Fluo from external applications (REST services, crawlers, etc).
```

Comment:
Another opinion,  feel free to ignore.  Thinking structuring this text so that Client and Observers come before Oracle and worker may be better. Basically introduce the user facing stuff before the internals.  Its not a big deal because the document is so short, a reader can easily hold all the info in memory.   I think the order would matter more if the document grows.